### ISSUE TITLE
Node Keeps Restarting, Won't Complete Blockchain Sync

### ISSUE THREAD

**Issue by jollypatron** [raspiblitz/raspiblitz#4805]
Hi Everyone,

I keep running into the same problem with my node. It starts its initial sync of the blockchain, and then after about 5-7% through, it restarts over and over and doesn't make any more progress.

<img width="321" alt="Screenshot 2024-10-16 192030" src="https://github.com/user-attachments/assets/6d0354f8-a100-491e-ace3-84b44a379851">

I've looked through some of the similar issues posted on the forum, but I haven't been able to find anything that works. I even tried starting my node again from scratch and I still get this message. Here is the debug log. Really appreciate any help you can provide!

***************************************************************
* RASPIBLITZ DEBUG LOGS 
***************************************************************
blitzversion: 1.11.1
chainnetwork: bitcoin / test
 03:35:27 up  7:41,  4 users,  load average: 1.44, 1.55, 1.57

*** FAILED SERVICES ***
list any servcies with problems: sudo systemctl list-units --failed
  UNIT                  LOAD   ACTIVE SUB    DESCRIPTION
● bitcoind.service      loaded failed failed Bitcoin daemon on mainnet
● smartmontools.service loaded failed failed Self Monitoring and Reporting Technology (SMART) Daemon

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.
2 loaded units listed.

*** SETUPPHASE / BOOTSTRAP ***
see logs: cat /home/admin/raspiblitz.log
setupPhase--> done
state--> ready

*** BACKGROUNDSERVICE ***
to monitor Background service call: sudo journalctl -f -u background

*** BLOCKCHAIN (MAINNET) SYSTEMD STATUS ***
× bitcoind.service - Bitcoin daemon on mainnet
     Loaded: loaded (/etc/systemd/system/bitcoind.service; enabled; preset: enabled)
     Active: failed (Result: exit-code) since Thu 2024-10-17 00:52:45 BST; 2h 42min ago
   Duration: 4h 58min 14.771s
    Process: 1932718 ExecStartPre=/home/admin/config.scripts/bitcoin.check.sh prestart mainnet (code=exited, status=0/SUCCESS)
    Process: 1932732 ExecStart=/usr/local/bin/bitcoind -daemonwait -conf=/mnt/hdd/bitcoin/bitcoin.conf -datadir=/mnt/hdd/bitcoin (code=exited, status=1/FAILURE)
        CPU: 141ms

Oct 17 00:52:45 raspberrypi systemd[1]: bitcoind.service: Failed with result 'exit-code'.
Oct 17 00:52:45 raspberrypi systemd[1]: Failed to start bitcoind.service - Bitcoin daemon on mainnet.

*** LAST BLOCKCHAIN (MAINNET) ERROR LOGS ***
sudo journalctl -u bitcoind -b --no-pager -n20
Oct 17 00:52:44 raspberrypi systemd[1]: bitcoind.service: Scheduled restart job, restart counter is at 5.
Oct 17 00:52:44 raspberrypi systemd[1]: Stopped bitcoind.service - Bitcoin daemon on mainnet.
Oct 17 00:52:44 raspberrypi systemd[1]: Starting bitcoind.service - Bitcoin daemon on mainnet...
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932719]: /bin/chgrp: changing group of '/mnt/hdd/bitcoin': Read-only file system
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932720]: sed: can't read /mnt/hdd/bitcoin/bitcoin.conf: Input/output error
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932721]: sed: can't read /mnt/hdd/bitcoin/bitcoin.conf: Input/output error
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932722]: grep: /mnt/hdd/bitcoin/bitcoin.conf: Input/output error
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932723]: sed: can't read /mnt/hdd/bitcoin/bitcoin.conf: Input/output error
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932724]: touch: cannot touch '/mnt/hdd/bitcoin/debug.log': Read-only file system
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932725]: chown: changing ownership of '/mnt/hdd/bitcoin/debug.log': Read-only file system
Oct 17 00:52:44 raspberrypi bitcoin.check.sh[1932726]: chmod: changing permissions of '/mnt/hdd/bitcoin/debug.log': Read-only file system
Oct 17 00:52:44 raspberrypi bitcoind[1932732]: Error: Error reading configuration file: specified config file "/mnt/hdd/bitcoin/bitcoin.conf" could not be opened.
Oct 17 00:52:44 raspberrypi systemd[1]: bitcoind.service: Control process exited, code=exited, status=1/FAILURE
Oct 17 00:52:44 raspberrypi systemd[1]: bitcoind.service: Failed with result 'exit-code'.
Oct 17 00:52:44 raspberrypi systemd[1]: Failed to start bitcoind.service - Bitcoin daemon on mainnet.
Oct 17 00:52:45 raspberrypi systemd[1]: bitcoind.service: Scheduled restart job, restart counter is at 6.
Oct 17 00:52:45 raspberrypi systemd[1]: Stopped bitcoind.service - Bitcoin daemon on mainnet.
Oct 17 00:52:45 raspberrypi systemd[1]: bitcoind.service: Start request repeated too quickly.
Oct 17 00:52:45 raspberrypi systemd[1]: bitcoind.service: Failed with result 'exit-code'.
Oct 17 00:52:45 raspberrypi systemd[1]: Failed to start bitcoind.service - Bitcoin daemon on mainnet.

*** LAST BLOCKCHAIN (MAINNET) INFO LOGS ***
sudo tail -n 50 /mnt/hdd/bitcoin/debug.log

*** LND (MAINNET) SYSTEMD STATUS ***
- OFF by config -

*** CORE LIGHTNING (MAINNET) SYSTEMD STATUS ***
- not activated -

*** BLOCKCHAIN (TESTNET) SYSTEMD STATUS ***
- OFF by config -
*** LND (TESTNET) SYSTEMD STATUS ***
- OFF by config -

*** CORE LIGHTNING (TESTNET) SYSTEMD STATUS ***
- not activated -

*** BLOCKCHAIN (SIGNET) SYSTEMD STATUS ***
- OFF by config -
*** LND (SIGNET) SYSTEMD STATUS ***
- OFF by config -

*** CORE LIGHTNING (SIGNET) SYSTEMD STATUS ***
- not activated -

*** NGINX SYSTEMD STATUS ***
● nginx.service - A high performance web server and a reverse proxy server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; preset: enabled)
    Drop-In: /etc/systemd/system/nginx.service.d
             └─raspiblitz.conf
     Active: active (running) since Wed 2024-10-16 19:54:15 BST; 7h ago
       Docs: man:nginx(8)
   Main PID: 2563 (nginx)
      Tasks: 5 (limit: 9438)
        CPU: 2.562s
     CGroup: /system.slice/nginx.service
             ├─2563 "nginx: master process /usr/sbin/nginx -g daemon on; master_process on;"
             ├─2565 "nginx: worker process"
             ├─2566 "nginx: worker process"
             ├─2567 "nginx: worker process"
             └─2568 "nginx: worker process"

Oct 16 19:54:14 raspberrypi systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Oct 16 19:54:15 raspberrypi systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.

*** LAST NGINX LOGS ***
sudo journalctl -u nginx -b --no-pager -n20
Oct 16 19:54:14 raspberrypi systemd[1]: Starting nginx.service - A high performance web server and a reverse proxy server...
Oct 16 19:54:15 raspberrypi systemd[1]: Started nginx.service - A high performance web server and a reverse proxy server.
--> CHECK CONFIG: sudo nginx -t
2024/10/17 03:35:28 [emerg] 2914829#2914829: cannot load certificate "/mnt/hdd/app-data/nginx/tls.cert": BIO_new_file() failed (SSL: error:80000005:system library::Input/output error:calling fopen(/mnt/hdd/app-data/nginx/tls.cert, r) error:10080002:BIO routines::system lib)
nginx: configuration file /etc/nginx/nginx.conf test failed

*** BLITZAPI STATUS ***
installed=1
repo='https://github.com/fusion44/blitz_api.git'
branch='blitz-v1.11'
commit='b08e53b599696edeeeaef1081caad6ad78900b48'
● blitzapi.service - BlitzBackendAPI
     Loaded: loaded (/etc/systemd/system/blitzapi.service; enabled; preset: enabled)
     Active: active (running) since Wed 2024-10-16 19:54:10 BST; 7h ago
   Main PID: 1082 (python)
      Tasks: 4 (limit: 9438)
        CPU: 2min 9.943s
     CGroup: /system.slice/blitzapi.service
             └─1082 /home/blitzapi/blitz_api/venv/bin/python -m uvicorn app.main:app --port 11111 --host=*.*.*.* --root-path /api

Oct 17 03:35:26 raspberrypi python[1082]: 2024-10-17 03:35:26 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:27 raspberrypi python[1082]: 2024-10-17 03:35:27 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.

*** LAST BLITZAPI LOGS ***
sudo journalctl -u blitzapi -b --no-pager -n20
Oct 17 03:35:08 raspberrypi python[1082]: 2024-10-17 03:35:08 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:09 raspberrypi python[1082]: 2024-10-17 03:35:09 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:10 raspberrypi python[1082]: 2024-10-17 03:35:10 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:11 raspberrypi python[1082]: 2024-10-17 03:35:11 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:12 raspberrypi python[1082]: 2024-10-17 03:35:12 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:13 raspberrypi python[1082]: 2024-10-17 03:35:13 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:14 raspberrypi python[1082]: 2024-10-17 03:35:14 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:15 raspberrypi python[1082]: 2024-10-17 03:35:15 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:16 raspberrypi python[1082]: 2024-10-17 03:35:16 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:17 raspberrypi python[1082]: 2024-10-17 03:35:17 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:18 raspberrypi python[1082]: 2024-10-17 03:35:18 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:19 raspberrypi python[1082]: 2024-10-17 03:35:19 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:20 raspberrypi python[1082]: 2024-10-17 03:35:20 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:21 raspberrypi python[1082]: 2024-10-17 03:35:21 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:22 raspberrypi python[1082]: 2024-10-17 03:35:22 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:23 raspberrypi python[1082]: 2024-10-17 03:35:23 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:24 raspberrypi python[1082]: 2024-10-17 03:35:24 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:25 raspberrypi python[1082]: 2024-10-17 03:35:25 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.
Oct 17 03:35:26 raspberrypi python[1082]: 2024-10-17 03:35:26 | ❌ | service.py:210 | Aiohttp client connection error: Cannot connect to host *.*.*.*:8332 ssl:default [Connect call failed ('127.0.0.1', 8332)]
Oct 17 03:35:27 raspberrypi python[1082]: 2024-10-17 03:35:27 | ℹ️ | utils.py:67 | Key 'tor_web_addr' not found in Redis DB.

*** BLITZ WebUI STATUS ***
installed=1
repo='https://github.com/raspiblitz/raspiblitz-web.git'
branch='release/v1.11'
commit='77dde492c2fdd8aebe1a38654a86a2911be67b20'

- TOUCHSCREEN is OFF by config
- Loop is OFF by config
- LND-RTL is OFF by config
- CL-RTL is OFF by config
- Electrum Rust Server is OFF by config
- LIT is OFF by config
- LNDg is OFF by config
* BTCPayServer is OFF by config
* BTC-RPC-Explorer is OFF by config
- LNbits is OFF by config
- Thunderhub is OFF by config
- SPECTER is OFF by config
- SPHINX is OFF by config
- FINTS is OFF by config

*** MOUNTED DRIVES ***
df -T -h
Filesystem     Type      Size  Used Avail Use% Mounted on
udev           devtmpfs  3.9G     0  3.9G   0% /dev
tmpfs          tmpfs     795M  1.4M  794M   1% /run
/dev/mmcblk0p2 ext4      117G   20G   93G  18% /
tmpfs          tmpfs     3.9G     0  3.9G   0% /dev/shm
tmpfs          tmpfs     5.0M     0  5.0M   0% /run/lock
tmpfs          tmpfs      32M   12K   32M   1% /var/cache/raspiblitz
/dev/mmcblk0p1 vfat      510M   64M  447M  13% /boot/firmware
/dev/sda1      ext4      1.8T   57G  1.7T   4% /mnt/hdd
tmpfs          tmpfs     795M   32K  795M   1% /run/user/1000
tmpfs          tmpfs     795M   32K  795M   1% /run/user/1001

*** SD CARD HOMES ***
sudo du -ahd1 /home
2.6G	/home/btcpay
349M	/home/lnbits
2.2G	/home/mempool
2.5G	/home/electrs
682M	/home/blitzapi
313M	/home/btcrpcexplorer
561M	/home/jam
629M	/home/joinmarket
811M	/home/bitcoin
464M	/home/admin
55M	/home/pi
392M	/home/rtl
1.4G	/home/thunderhub
13G	/home

*** LOGFILES ***
Archived and active journals take up 67.5M in the file system.
84M	/var/log

*** DATADRIVE ***
# raspi_bootdir(/boot/firmware)
# RASPIBLITZ DATA DRIVE Status

# BASICS
isMounted=1
isBTRFS=0
hddRaspiData=1
hddRaspiVersion=''
isSMART=0
isSSD=1
datadisk='sda'
datapartition='sda1'
hddCandidate='sda'
hddPartitionCandidate='sda1'
hddTemperature=
hddBlocksBitcoin=1
hddBytes=2000350494720
hddGigaBytes=1862
hddTBSize='2TB'
hddUsedInfo='2TB 4%'
hddDataFreeBytes=1806917312512
hddDataFreeKB=1764567688
hddDataFreeGB=1682
hddAdapterUSB=''
hddAdapterUSAP=0

# RAID
isRaid=0
raidCandidates=0

# SWAP
isSwapExternal=1
SwapExternalPath='/mnt/hdd/swapfile'

smartctl 7.3 2022-02-28 r5338 [aarch64-linux-6.6.20+rpt-rpi-v8] (local build)
Copyright (C) 2002-22, Bruce Allen, Christian Franke, www.smartmontools.org

/dev/sda: Unknown USB bridge [0x0bda:0x9201 (0xf200)]
Please specify device type with the -d option.

Use smartctl -h to get a usage summary

*** NETWORK ***
localip=*.*.*.*
localiprange=*.*.*.*/24
dhcp=1
network_device=eth0

*** ZRAM ***
downloaded=1
serviceLoaded=1
serviceActive=1

*** HARDWARE TEST RESULTS ***
throttled=0x0
UndervoltageReports in Logs: 0

*** SYSTEM CACHE STATUS ***
system_count_start_blockchain="6"
system_up_text="7h"
system_cpu_load=" 1.44, 1.55, 1.57"
system_ups_battery=""
system_temp_fahrenheit="149"
system_temp_celsius="65"
system_ups_status="OFF"
system_ram_mb="7945"
system_ram_available_mb="7477"
system_count_undervoltage="0"
system_up="27698"
system_board="rp5"
system_ram_gb="8"
system_init_time="1729104851"
ln_default_sync_initial_done=""
btc_default_mempool_transactions="0"
btc_default_sync_initialblockdownload="1"
btc_default_sync_progress="0.07498573968822972"
btc_default_blocks_behind="494319"
btc_default_sync_percentage="7.49"
btc_default_peers="10"
btc_default_running="0"
btc_default_sync_initial_done="0"
btc_default_blocks_data_kb="48582356"
btc_default_online="0"
btc_default_blocks_headers="865968"
btc_default_port="8333"
btc_default_ready="0"
btc_default_synced="0"
btc_default_blocks_verified="371649"
btc_default_version="v26.0.0"
btc_default_error_short=""
btc_default_activated="1"
btc_default_error_full=""
*** POSSIBLE ERROR REPORTS ***

**Comment by mrpaper87** [comment]
Same problem:
![image](https://github.com/user-attachments/assets/ec45b83c-714c-429d-baa8-e64dade6aa90)

**Comment by jollypatron** [comment]
I did some more digging, and I found that it might be an issue with the amount of power that is being delivered to my SSD via the raspberry PI. If not enough power is getting to SSD, it can impact its ability to read/write. This can especially happen if you're not using the recommended SSD from the documentation (which I am not).

I started the entire process over again (re-flashed micro SD). I also went out and got a USB power hub so that the SSD has its own power source and it's not solely reliant on the raspberry pi. This is the one I got on Amazon: https://www.amazon.com/dp/B00TPMEOYM?ref=ppx_yo2ov_dt_b_fed_asin_title&th=1

Good news is that it seems to be working now! Sync is still in progress and not complete yet, but it's made it a lot further than it normally does. Will update this thread if it completes properly.

I've seen a lot of similar posts on here of people with the same issue, and it's not always a power issue. But I think this is what it was for me...hopefully this helps!
